users/tests.py

```diff
@@ -1,3 +1,157 @@
+        make_profile(username=PROFIL_INFORMATION.get('username')[2],
+                     password=PROFIL_INFORMATION.get('password')[2],
+                     city='Toronto',
+                     email='john2.doe@example.com',
+                     phone_number='0541234636')
+        make_profile(username=PROFIL_INFORMATION.get('username')[0],
+                     password=PROFIL_INFORMATION.get('password')[0],
+                     city='London',
+                     email='john3.doe@example.com',
+                     phone_number='0541234563')
+
+        assert list(Profile.filter_by_city(PROFIL_INFORMATION.get('city')[0])) == [new_profile]
+
+    def test_filter_by_first_name(self, make_profile):
+        new_profile = make_profile(username=PROFIL_INFORMATION.get('username')[1],
+                                   password=PROFIL_INFORMATION.get('password')[1],
+                                   email=PROFIL_INFORMATION.get('email')[1],
+                                   phone_number=PROFIL_INFORMATION.get('phone_number')[1])
+        make_profile(username=PROFIL_INFORMATION.get('username')[2],
+                     password=PROFIL_INFORMATION.get('password')[2],
+                     first_name='David',
+                     email=PROFIL_INFORMATION.get('email')[2],
+                     phone_number=PROFIL_INFORMATION.get('phone_number')[2])
+        make_profile(username=PROFIL_INFORMATION.get('username')[0],
+                     password=PROFIL_INFORMATION.get('password')[0],
+                     first_name='David',
+                     email=PROFIL_INFORMATION.get('email')[0],
+                     phone_number=PROFIL_INFORMATION.get('phone_number')[0])
+
+        assert list(Profile.filter_by_first_name(PROFIL_INFORMATION.get('first_name')[0])) == [new_profile]
+
+    def test_filter_by_last_name(self, make_profile):
+        new_profile = make_profile(username=PROFIL_INFORMATION.get('username')[1],
+                                   password=PROFIL_INFORMATION.get('password')[1],
+                                   email=PROFIL_INFORMATION.get('email')[1],
+                                   phone_number=PROFIL_INFORMATION.get('phone_number')[1])
+        make_profile(username=PROFIL_INFORMATION.get('username')[2],
+                     password=PROFIL_INFORMATION.get('password')[2],
+                     last_name='David',
+                     email=PROFIL_INFORMATION.get('email')[2],
+                     phone_number=PROFIL_INFORMATION.get('phone_number')[2])
+        make_profile(username=PROFIL_INFORMATION.get('username')[0],
+                     password=PROFIL_INFORMATION.get('password')[0],
+                     last_name='David',
+                     email=PROFIL_INFORMATION.get('email')[0],
+                     phone_number=PROFIL_INFORMATION.get('phone_number')[0])
+
+        assert list(Profile.filter_by_last_name(PROFIL_INFORMATION.get('last_name')[0])) == [new_profile]
```

Comment:
same. 
also if the tests are that similar, I would use a pytest.mark.parametrize

Reply:
I don't know this feature unfortunately

Reply:
I worked with factory data(I think its called that)

Reply:
You can read about that here https://docs.pytest.org/en/6.2.x/parametrize.html
or ask @idan1356  , he implemented it (with a factory) beautifully.

Reply:
So I would appreciate your help
 @idan1356

Reply:
Sure, lets discuss it tomorrow

Reply:
fixed